posthog/tasks/update_cache.py

```diff
@@ -60,12 +83,36 @@ def update_cached_items() -> None:
     taskset.apply_async()
 
 
-def _calculate_trends(filter: Filter, key: str, team_id: int) -> List[Dict[str, Any]]:
-    actions = Action.objects.filter(team_id=team_id)
-    actions = actions.prefetch_related(Prefetch("steps", queryset=ActionStep.objects.order_by("id")))
+def get_cache_type(filter: Filter) -> CacheType:
+    if filter.insight == FUNNELS:
+        return CacheType.FUNNEL
+    elif filter.insight == PATHS:
+        return CacheType.PATHS
+    elif filter.insight == RETENTION:
+        return CacheType.RETENTION
+    elif filter.insight == TRENDS and filter.shown_as == STICKINESS:
+        return CacheType.STICKINESS
+    else:
+        return CacheType.TRENDS
+
+
+def import_from(module: str, name: str) -> Any:
+    return getattr(importlib.import_module(module), name)
+
+
+def _calculate_by_filter(
```

Comment:
Couple things about this file
1. It doesn't get called on ee yet in posthog/celery.py, so we'll need to change the conditional there
2. I'm worried this will again completely clog clickhouse queries. I had a slightly better solution for cohorts (once every minute, only do a handful at a time) but it's still not perfect.
3. Be good to have tests

Reply:
Got it! Thanks for the heads up. I have a weird habit of marking the PR as ready for review when an initial implementation is in but didn't finish tests yet sorry for premature request if that's what it seemed like